libraries/AP_Motors/AP_Motors_Class.cpp

```diff
@@ -18,20 +18,63 @@
 #include <SRV_Channel/SRV_Channel.h>
 #include <GCS_MAVLink/GCS.h>
 #include <AP_Notify/AP_Notify.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
+
+// this attempts to capture old default defines, but is not fool proof as it cannot see the
+// old vehicle level defines that may have been added to defines.h,
+// but will work with defines from hwdef
+#ifndef PWM_RATE_DEFAULT
+    #if APM_BUILD_TYPE(APM_BUILD_ArduCopter) || APM_BUILD_TYPE(APM_BUILD_Blimp)
+        #ifdef RC_FAST_SPEED
+            #define PWM_RATE_DEFAULT RC_FAST_SPEED
+        #endif
+
+    #elif APM_BUILD_TYPE(APM_BUILD_Heli)
+        #define PWM_RATE_DEFAULT 125
+
+    #elif APM_BUILD_TYPE(APM_BUILD_ArduSub)
+        #ifdef RC_SPEED_DEFAULT
+            #define PWM_RATE_DEFAULT RC_SPEED_DEFAULT
+        #else
+            #define PWM_RATE_DEFAULT 200
+        #endif
+
+    #endif
+
+    #ifndef PWM_RATE_DEFAULT
+        #define PWM_RATE_DEFAULT 490
+    #endif
+#endif
+
+const AP_Param::GroupInfo AP_Motors::var_info[] = {
+
+    // @Param: PWM_RATE
+    // @DisplayName: ESC Update rate
+    // @Description: This is the rate in Hertz that your ESCs will receive updates
+    // @Units: Hz
+    // @Range: 50 490
```

Comment:
isnt this range actually 50 to 20Khz?

Reply:
I just moved it, the valid range depends on your PWM type.

Reply:
we should change it.....20K is allowed on brushed motor controllers....this requires an override presently for those applications....might as well fix it now

Reply:
Larger than 400 only works with brushed PWM type. The true upper limit is a little more complex to work out, I guess it depends on what clock speed your running and how the DMA is setup per board.

Reply:
Basically I think brushed is a such a rarely used feature that were better to keep the range set to what works for normal PWM.

Reply:
so we should restrict the range to 100hz only because some brushless ESCs will not work much above 50Hz?...I think its not a valid argument myself....we have many params that if the user sets the param wrong for his hardware, there will be problems

Reply:
the brushless motor controller wiki page recommends 16K with 20K being the max....if that is incorrect for certain conditions, we should note it.... @rmackay9 ?

Reply:
we discussed and decided to make wiki notes about how to get the param changed for brushless controllers for out of range values...